### PR TITLE
Download uri inside coroutine scope

### PR DESCRIPTION
**Pull request by ronnieotieno**

### CODE REVIEW

File: intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService2.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        url?.let {
+            job?.cancel()
+            job = CoroutineScope(Dispatchers.IO).launch {
+                try {
+                    val result = download(url)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
+            }
+        }
```
Comment (samuelowino) [inline intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService2.kt:42]: I didn't expect that you could construct a Job from a CoroutineScope. 
very nice 👍🏿

File: intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
+import androidx.lifecycle.LifecycleService
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import java.net.HttpURLConnection
 import java.net.URL
 import java.nio.charset.Charset
 
-class DownloadService : Service() {
+class DownloadService : LifecycleService() {
```
Comment (samuelowino) [inline intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt:15]: LifecycleService seems more promising, I just wonder if you can still launch it with ```startService(intent)```?

Comment (ronnieotieno) [reply intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt:15]: Yes absolutely!

File: intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt [hunk 3]
```diff
@@ -25,22 +30,24 @@ class DownloadService : Service() {
         }
         Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
         url?.let {
-            download(url)
+            lifecycleScope.launch(Dispatchers.IO) {
+                val result = download(url)
+            }
```
Comment (samuelowino) [inline intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt:35]: This explains it

**Pull request merged**